AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -1491,13 +1469,116 @@ - (void)measureNodeWithBoundsIfNecessary:(CGRect)bounds
+    CGSize parentSize = bounds.size;
+    ASSizeRange constrainedSize = [self constrainedSizeForLayoutPass];
+    
+    return std::make_shared<ASDisplayNodeLayout>(
+      [self calculateLayoutThatFits:constrainedSize restrictedToSize:self.style.size relativeToParentSize:parentSize],
+      constrainedSize,
+      parentSize
+    );
+  }();
+  
+  // If the size of the new layout to apply did change from the current bounds, invalidate the whole tree up
+  // so the root node can handle a resizing if necessary
+  if (CGSizeEqualToSize(ASCeilSizeValues(bounds.size), pendingLayout->layout.size) == NO) {
+    [self invalidateSize];
+  }
+
+  if (didCreateNewContext) {
+    ASLayoutElementClearCurrentContext();
+  } else if (didOverrideExistingContext) {
+    context.needsVisualizeNode = !context.needsVisualizeNode;
+    ASLayoutElementSetCurrentContext(context);
+  }
+
+  // Finally transition to pendingLayout
+  ASDisplayNodeAssertNotNil(pendingLayout->layout, @"pendintLayout->layout should not be nil! %@", self);
+
+  _pendingLayoutTransition = [[ASLayoutTransition alloc] initWithNode:self
+                                                        pendingLayout:pendingLayout
+                                                       previousLayout:previousLayout];
+
+  // Only complete the pending layout transition if the node is not a subnode of a node that is currently
+  // in a layout transition
+  if (ASHierarchyStateIncludesLayoutPending(_hierarchyState) == NO) {
+    // Complete the pending layout transition immediately
+    [self _completePendingLayoutTransition];
+  }
+}
+
+- (ASSizeRange)constrainedSizeForLayoutPass
+{
+  // Use as default constrained size the bounds
+  CGRect bounds = self.threadSafeBounds;
+  ASSizeRange constrainedSize = ASSizeRangeMake(bounds.size);
+  
+  // Checkout if constrained size of pending or calculated display node layout can be used
+  if (_pendingDisplayNodeLayout != nullptr && CGSizeEqualToSize(_pendingDisplayNodeLayout->layout.size, ASCeilSizeValues(bounds.size))) {
+    // We assume the size from the last returned layoutThatFits: layout was applied so use the pejnding display node
+    // layout constrained size
+    constrainedSize = _pendingDisplayNodeLayout->constrainedSize;
+  } else if (CGSizeEqualToSize(_calculatedDisplayNodeLayout->layout.size, CGSizeZero) == NO &&
+             CGSizeEqualToSize(_calculatedDisplayNodeLayout->layout.size, ASCeilSizeValues(bounds.size))) {
+    // We assume the  _calculatedDisplayNodeLayout is still valid and the frame is not different
+    constrainedSize = _calculatedDisplayNodeLayout->constrainedSize;
+  } else {
+    // In this case neither the _pendingDisplayNodeLayout or the _calculatedDisplayNodeLayout constrained size can
+    // be reused, so the current bounds is used. This is usual the case if a frame was set manually that differs to
+    // the one returned from layoutThatFits: or layoutThatFits: was never called
```

Comment:
Let's just put the return in each of the if / else if / else.

That way, we don't need the constrainedSize variable at all — and then this else condition will clearly show it is return ASSizeRangeMake(bounds.size);